utils/src/test/scala/za/co/absa/enceladus/utils/types/TypedStructFieldSuite.scala

```diff
@@ -15,21 +30,24 @@ class TypedStructFieldSuite extends FunSuite {
   private def createField(dataType: DataType,
                           nullable: Boolean = false,
                           default: Option[Any] = None,
-                          otherMetadata: Map[String, String] = Map.empty
+                          otherMetadata: Map[String, Any] = Map.empty
                          ): StructField = {
-    val medatadaBuilder: MetadataBuilder = otherMetadata.foldLeft(new MetadataBuilder()) (
-      (builder, data) => builder.putString(data._1, data._2)
-    )
-    val metadata = (default match {
-      case None             => medatadaBuilder
-      case Some(null)       => medatadaBuilder.putNull(MetadataKeys.DefaultValue) // scalastyle:ignore null
-      case Some(s: String)  => medatadaBuilder.putString(MetadataKeys.DefaultValue, s)
-      case Some(i: Int)    => medatadaBuilder.putLong(MetadataKeys.DefaultValue, i)
-      case Some(l: Long)    => medatadaBuilder.putLong(MetadataKeys.DefaultValue, l)
-      case Some(b: Boolean) => medatadaBuilder.putBoolean(MetadataKeys.DefaultValue, b)
-      case Some(d: Double)  => medatadaBuilder.putDouble(MetadataKeys.DefaultValue, d)
-      case Some(x)          => medatadaBuilder.putString(MetadataKeys.DefaultValue, x.toString)
-    }).build()
+    def addMetadata(builder: MetadataBuilder, key: String, value: Option[Any]): MetadataBuilder = {
+      value match {
+        case None => builder
+        case Some(null) => builder.putNull(key) // scalastyle:ignore null
+        case Some(s: String) => builder.putString(key, s)
+        case Some(i: Int) => builder.putLong(key, i)
+        case Some(l: Long) => builder.putLong(key, l)
+        case Some(b: Boolean) => builder.putBoolean(key, b)
+        case Some(f: Float) => builder.putDouble(key, f)
+        case Some(d: Double) => builder.putDouble(key, d)
+        case Some(x) => builder.putString(key, x.toString)
+      }
+    }
+    val metadataBuilder: MetadataBuilder = otherMetadata.foldLeft(new MetadataBuilder()) (
+      (builder, data) => addMetadata(builder, data._1, Some(data._2)))
```

Comment:
Consider replacing `._1` and `._2` with matching (unapplying) the pair and give names to its elements.

Reply:
Hope the solution with named parameters address the issue well enough. I don't like using the numeric designation of tuple members, but here I considered it good trade-off for more concise code.